plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheck.kt

```diff
@@ -0,0 +1,75 @@
+package org.jlleitschuh.gradle.ktlint
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileTree
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Console
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import org.jlleitschuh.gradle.ktlint.reporter.KtlintReport
+import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+
+open class KtlintCheck : DefaultTask() {
```

Comment:
:o: Implementing `VerificationTask` would make sense

Reply:
Implementing `VerificationTask` would somewhat force me to drop the `Property<Boolean>` type for `ignoreFailures`. Then I would need to use convention mapping, right?

Reply:
What about `SourceTask`?

Reply:
About `SourceTask`, the extra use case of including/excluding individual files could be tackled later, as a separate PR.

About `VerificationTask`, that's unfortunate but I see your point and agree @wolfs